From a77458fe8a7276fcd72c501d4de6007eafbaacb9 Mon Sep 17 00:00:00 2001
From: Marcelo Tosatti <mtosatti@redhat.com>
Date: Tue, 1 Feb 2011 18:29:17 -0200
Subject: [PATCH 20/24] PATCH RHEL6.1 qemu-kvm vga: tag as not hotplugable (v3)

RH-Author: Marcelo Tosatti <mtosatti@redhat.com>
Message-id: <20110201182917.GA19465@amt.cnet>
Patchwork-id: 17429
O-Subject: [patch 4/6] PATCH RHEL6.1 qemu-kvm vga: tag as not hotplugable (v3)
Bugzilla: 498774
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

patch from v2 series had a fuzz, probably the reason for being declared
broken. Rebased patch attached.

---
vga: tag as not hotplugable.

From: Gerd Hoffmann <kraxel@redhat.com>

This patch tags all vga cards as not hotpluggable.  The qemu
standard vga will never ever be hotpluggable.  For cirrus + vmware
it might be possible to get that work some day.  Todays we can't
handle that for a number of reasons though.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
(upstream backport of commit be92bbf73dfd7d8a4786dc5f6c71590f4fbc5a32)
BZ: 498774

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 hw/cirrus_vga.c |    1 +
 hw/vga-pci.c    |    1 +
 hw/vmware_vga.c |    1 +
 3 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/hw/cirrus_vga.c b/hw/cirrus_vga.c
index 21d057a..d04adeb 100644
--- a/hw/cirrus_vga.c
+++ b/hw/cirrus_vga.c
@@ -3242,6 +3242,7 @@ static PCIDeviceInfo cirrus_vga_info = {
     .qdev.desc    = "Cirrus CLGD 54xx VGA",
     .qdev.size    = sizeof(PCICirrusVGAState),
     .qdev.vmsd    = &vmstate_pci_cirrus_vga,
+    .no_hotplug   = 1,
     .init         = pci_cirrus_vga_initfn,
     .romfile      = VGABIOS_CIRRUS_FILENAME,
     .config_write = pci_cirrus_write_config,
diff --git a/hw/vga-pci.c b/hw/vga-pci.c
index c422c91..f6fb1b3 100644
--- a/hw/vga-pci.c
+++ b/hw/vga-pci.c
@@ -136,6 +136,7 @@ static PCIDeviceInfo vga_info = {
     .qdev.name    = "VGA",
     .qdev.size    = sizeof(PCIVGAState),
     .qdev.vmsd    = &vmstate_vga_pci,
+    .no_hotplug   = 1,
     .init         = pci_vga_initfn,
     .config_write = pci_vga_write_config,
     .qdev.props   = (Property[]) {
diff --git a/hw/vmware_vga.c b/hw/vmware_vga.c
index 0655587..a9767a0 100644
--- a/hw/vmware_vga.c
+++ b/hw/vmware_vga.c
@@ -1272,6 +1272,7 @@ static PCIDeviceInfo vmsvga_info = {
     .qdev.name    = "vmware-svga",
     .qdev.size    = sizeof(struct pci_vmsvga_state_s),
     .qdev.vmsd    = &vmstate_vmware_vga,
+    .no_hotplug   = 1,
     .init         = pci_vmsvga_initfn,
     .romfile      = "vgabios-vmware.bin",
 };
-- 
1.7.4.rc1.16.gd2f15e

